File: dream/lib/cards_project/marketplace/order_events.ml
(* Domain events for Order *)

type order_paid_t = {
  order_id : int;
  player_id : int;
  total : float;
  payment_method : string;
  paid_at : string;
} [@@deriving yojson]

type order_shipped_t = {
  order_id : int;
  tracking_number : string;
  shipped_at : string;
} [@@deriving yojson]

type order_refunded_t = {
  order_id : int;
  refunded_at : string;
} [@@deriving yojson]

File: gin/internal/model/marketplace/order_events.go
package model_marketplace

import (
	"cards_project/internal/types"
)

// Domain events emitted by Order.
type OrderOrderPaidEvent struct {
	OrderId int `json:"order_id"`
	PlayerId int `json:"player_id"`
	Total types.Decimal `json:"total"`
	PaymentMethod string `json:"payment_method"`
	PaidAt string `json:"paid_at"`
}

type OrderOrderShippedEvent struct {
	OrderId int `json:"order_id"`
	TrackingNumber string `json:"tracking_number"`
	ShippedAt string `json:"shipped_at"`
}

type OrderOrderRefundedEvent struct {
	OrderId int `json:"order_id"`
	RefundedAt string `json:"refunded_at"`
}
